source/extensions/common/dynamic_forward_proxy/dns_cache_impl.cc

```diff
@@ -90,10 +90,20 @@ absl::StatusOr<Network::DnsResolverSharedPtr> DnsCacheImpl::selectDnsResolver(
     Event::Dispatcher& main_thread_dispatcher,
     Server::Configuration::CommonFactoryContext& context) {
   envoy::config::core::v3::TypedExtensionConfig typed_dns_resolver_config;
-  Network::DnsResolverFactory& dns_resolver_factory =
-      Network::createDnsResolverFactoryFromProto(config, typed_dns_resolver_config);
-  return dns_resolver_factory.createDnsResolver(main_thread_dispatcher, context.api(),
-                                                typed_dns_resolver_config);
+  Network::DnsResolverFactory* dns_resolver_factory;
+
+  if (!config.use_tcp_for_dns_lookups() && !config.has_typed_dns_resolver_config() &&
```

Comment:
Why does `use_tcp_for_dns_lookups` need to be `false`?

Also can you add a comment about this logic here, please? It is difficult to understand from the code itself.

Reply:
done